src/routes/customers.php

```diff
@@ -15,8 +15,147 @@
         $stmt = $db->query($sql);
         $customers = $stmt->fetchAll(PDO::FETCH_OBJ);
         $db = null;
-        echo json_encode($customers);
+        $response->write(json_encode($customers));
+        return $response->withStatus(200);
     } catch (PDOException $e) {
-        echo '{"error": {"text": ' . $e->getMessage() . '}';
+        $db = null;
+        $response->write($e);
+        return $response->withStatus(500);
+    }
+});
+
+// Get Customer By Username
+$app->get('/api/customers/{username}', function (Request $request, Response $response, array $args) {
+    $username = $args['username'];
+    $sql = "SELECT * FROM customers WHERE username = :username";
+
+    try {
+        // Get DB Object
+        $db = new db();
+
+        $db = $db->connect();
+        $stmt = $db->prepare($sql);
+        $stmt->bindParam(':username', $username);
+        $stmt->execute();
+        $customer = $stmt->fetch(PDO::FETCH_OBJ);
+        $db = null;
+
+        $response->write(json_encode($customer));
+        return $response->withStatus(200);
+    } catch (PDOException $e) {
+        $db = null;
+        $response->write($e);
+        return $response->withStatus(500);
+    }
+});
+
+// Login for customers
+$app->post('/api/customers/login', function (Request $request, Response $response) {
+
+    $username = $request->getParam('username');
+    $password = $request->getParam('password');
+
+    $sql = "SELECT * FROM customers where username = '$username'";
+
+    try {
+        // Get DB Object
+        $db = new db();
+        // Connect
+        $db = $db->connect();
+        $stmt = $db->query($sql);
+
+        $customer = $stmt->fetch(PDO::FETCH_OBJ);
+        $db = null;
+
+        if (!$customer) {
+            return $response->write('no admin found')->withStatus(404);
+        }
+
+        if ($customer->password == $password) {
+            //SUCCESS
+            $responseArray = array();
+            $responseArray['userType'] = 'customer';
+            return $response->write(json_encode($responseArray));
+            return $response->withStatus(200);
+        } else {
+            //FAILED
+            return $response->write('wrong credential')->withStatus(409);
+        }
+    } catch (PDOException $e) {
+        $db->rollBack();
```

Comment:
you don't need rollback here, also check hotel.php code for the correct usage of rollback function